app/models/king.rb

```diff
@@ -13,6 +17,35 @@ def valid_move?(x,y)
     return false
   end
 
+
+  def can_castle?(x,y)
+    updated_at == created_at && \
+    ((x == 6 && castle_rook && castle_rook.x_position == 7 && \
+    castle_rook.updated_at == castle_rook.created_at && \
+    castle_rook.path_blocked?(5, y_position) == false) || \
+    (x == 2 && castle_rook && castle_rook.x_position == 0 && \
+    castle_rook.updated_at == castle_rook.created_at && \
+    castle_rook.path_blocked?(3, y_position) == false )) && \
+    [y, castle_rook.y_position].uniq == y_position
+  end
+
+
+  def castle_rook
+    @castle_rook = game.pieces.where(x_position: 0, type: Rook).first || game.pieces.where(x_position: 7, type: Rook).first
```

Comment:
I had it as 'Rook' at first, but it threw an error "unknown method or variable".  I'll try it again and double check.  
On def castle_rook, my thinking was that even though it would return any rook that might be in that x position, the other checks would narrow that down to the correct rook via "updated at = created at" confirming that the rook hadn't moved, and the y-position criteria verifying that it was in the same row as the king being moved.  Guess I had the right idea, just wasn't quite there!  I'll refactor with your suggestions and continue testing.  Thanks!